### PR TITLE
Fixed test_inverse and test_logdet

### PR DESCRIPTION
**Pull request by AndPotap**

### CODE REVIEW

File: cola/algorithms/slq.py [hunk 1]
```diff
@@ -39,11 +39,12 @@ def slq_fwd(A, fun, num_samples, max_iters, tol, pbar, key):
     xnp = A.xnp
     rhs = xnp.randn(A.shape[1], num_samples, dtype=A.dtype, key=key, device=A.device)
     alpha, beta, _, iters, _ = lanczos_parts(A, rhs, max_iters, tol, pbar)
-    if xnp.__name__.find("torch") >= 0:
-        alpha, beta = alpha[..., :iters - 1], beta[..., :iters]
-    # REMOVED by marc due to it breaking jit compilation
-    else:
-        alpha = alpha[..., :-1]
+    # if xnp.__name__.find("torch") >= 0:
```
Comment (AndPotap) [inline cola/algorithms/slq.py]: @mfinzi , this logic of not trimming the coefficients was resulting in the log() function being evaluated in zeros for the test_logdet.

File: tests/linalg/operator_market.py [hunk 2]
```diff
@@ -18,7 +18,7 @@
     'psd_scalarmul',
     'selfadj_hessian',
     'selfadj_tridiagonal',
-    'square_big',  # skipped by default
+    # 'square_big',  # skipped by default
```
Comment (AndPotap) [inline tests/linalg/operator_market.py:21]: @mfinzi , I'm gonna move this case outside of this unit tests

Comment (mfinzi) [reply tests/linalg/operator_market.py:21]: sounds good

File: cola/algorithms/slq.py (hunk 1 above)
Comment (mfinzi) [reply cola/algorithms/slq.py]: that makes sense, but how do we fix it given that the function may need to be jittable?

Comment (mfinzi) [reply cola/algorithms/slq.py]: and even disregarding jit, won't we have the same problem if the batched lanczos procedure terminates after some alphas are 0 for some of the probe vectors?

Comment (mfinzi) [reply cola/algorithms/slq.py:42]: instead what if we add a
`log_eigvals = xnp.where(eigvals>1e-7*xnp.max(eigvals), fun(eigvals), xnp.zeros_like(eigvals)`
then those will not be included in the sum

File: tests/linalg/operator_market.py (hunk 2 above)
Comment (mfinzi) [reply tests/linalg/operator_market.py:21]: I will add a decorator tomorrow to manage these tricky test cases

File: cola/algorithms/slq.py (hunk 1 above)
Comment (mfinzi) [reply cola/algorithms/slq.py]: instead what if we add a
log_eigvals = xnp.where(eigvals>1e-6*xnp.max(eigvals), fun(eigvals), xnp.zeros_like(eigvals))
then those will not be included in the sum

And replace 1e-6 with 10*xnp.get_machine_eps(dtype)

**Review by mfinzi — CHANGES_REQUESTED** [review]

File: cola/algorithms/slq.py (hunk 1 above)
Comment (AndPotap) [reply cola/algorithms/slq.py]: I changed this to your suggestion with a function that retrieves the machine precision values given the dtype

File: cola/algorithms/slq.py
Comment (mfinzi) [inline cola/algorithms/slq.py:1]: I would probably leave a little bit of wiggle room, maybe 10x that value as the cutoff

File: cola/jax_fns.py
Comment (mfinzi) [inline cola/jax_fns.py:1]: I think you want
np.finfo(dtype).eps
torch.finfo(dtype).eps

**Review by mfinzi — COMMENTED** [review]
I'll make a couple changes and submut

**Review by mfinzi — APPROVED** [review]

**Pull request merged**